lib/evss/service.rb

```diff
@@ -4,10 +4,24 @@
 
 module EVSS
   class Service < Common::Client::Base
+    STATSD_KEY_PREFIX = 'api.evss'
+
+    protected
+
     def headers_for_user(user)
       EVSS::AuthHeaders.new(user).to_h
     end
 
+    def with_monitoring
+      caller = caller_locations(1, 1)[0].label
+      yield
+    rescue StandardError => error
+      StatsD.increment("#{STATSD_KEY_PREFIX}.#{caller}.fail", tags: ["error:#{error.class}"])
```

Comment:
Is the `tags` logic of _get-the-exception-class-name_ fine enough?  Given a `Common::Client::Errors::ClientError`, it looks like it can have 3 separate meanings:

```ruby
when Common::Client::Errors::ClientError
  raise Common::Exceptions::Forbidden if error.status == 403
  log_message_to_sentry(error.message, :error, extra_context: { url: config.base_path, body: error.body })
  raise_backend_exception('EVSS400', self.class, error) if error.status == 400
  raise_backend_exception('EVSS502', self.class, error)
```

Reply:
Another thought, would it be desirable to reverse the order?  Call `handle_error` first and use `StatsD` to increment those exceptions?

Reply:
@patrickvinograd @wyattwalter we're incrementing failures and the total for each evss service action (download letter etc) right now a failure increment key is generic `api.evss.download_letter.fail` but it also gets tagged with the error that caused it. Is that sufficient to create a stacked graph? (errors stacked to some total over total calls)

Reply:
Yes, the tag values will get turned into slice-able label values for the time series data in prometheus.